src/overlays/actors/ovl_En_Az/z_en_az.c

```diff
@@ -39,108 +80,1758 @@ const ActorInit En_Az_InitVars = {
+                }
+            } else {
+                if ((this->unk_2FA == 0) && !(this->unk_374 & 2)) {
+                    Player* player = GET_PLAYER(play);
+                    Vec3f sp38;
+                    s16 temp_a1_2 = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+                    s16 temp_v0_7;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97C4C.s")
+                    if (ABS(temp_a1_2) < 0x1800) {
+                        Math_SmoothStepToS(&this->unk_3D4, temp_a1_2, 2, 0x71C, 0);
+                    } else {
+                        Math_SmoothStepToS(&this->unk_3D4, 0, 2, 0x71C, 0);
+                    }
+                    Math_Vec3f_Copy(&sp38, &player->actor.world.pos);
+                    sp38.y = player->bodyPartsPos[7].y + 3.0f;
+                    temp_v0_7 = Math_Vec3f_Pitch(&this->actor.focus.pos, &sp38);
+                    if (ABS(temp_v0_7) < 0x800) {
+                        Math_SmoothStepToS(&this->unk_3D6, temp_v0_7, 3, 0x71C, 0);
+                    } else {
+                        Math_SmoothStepToS(&this->unk_3D6, 0, 3, 0x71C, 0);
+                    }
+                }
+                if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+                    func_80A97114(this, play);
+                    this->unk_378 = 2;
+                    if ((this->unk_3D2 == 0x10CE) || (this->unk_3D2 == 0x10D4)) {
+                        this->unk_378 = 9;
+                    }
+                } else {
+                    Actor_GetScreenPos(play, &this->actor, &sp56, &sp54);
+                    if ((sp56 >= 0) && (sp56 <= SCREEN_WIDTH) && (sp54 >= 0) && (sp54 <= SCREEN_HEIGHT) &&
+                        func_800B8500(&this->actor, play, 120.0f, 120.0f, 0)) {
+                        this->unk_3D2 = func_80A97274(this, play);
+                        if ((this->unk_3D2 == 0x10CE) || (this->unk_3D2 == 0x10D4)) {
+                            this->actor.textId = 0;
+                        } else {
+                            this->actor.textId = this->unk_3D2;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void func_80A979DC(EnAz* this, PlayState* play) {
+    this->actionFunc = func_80A979F4;
+}
+
+void func_80A979F4(EnAz* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void func_80A97A28(EnAz* this, PlayState* play) {
+    this->actionFunc = func_80A97A40;
+}
+
+void func_80A97A40(EnAz* this, PlayState* play) {
+    if (SubS_StartActorCutscene(&this->actor, 0, -1, 0)) {
+        play->msgCtx.msgMode = 0;
+        play->msgCtx.msgLength = 0;
+        func_80A97A9C(this, play);
+    }
+}
+
+void func_80A97A9C(EnAz* this, PlayState* play) {
+    this->actionFunc = func_80A97AB4;
+}
+
+void func_80A97AB4(EnAz* this, PlayState* play) {
+    switch (Message_GetState(&play->msgCtx)) {
+        case 0:
```

Comment:
We just got the TalkState enum in z64message.h for the whatever gets returned from `Message_GetState`, if you could use that for the switch cases here.